src/metrics/histogram.rs

```diff
@@ -122,6 +122,27 @@ pub fn exponential_buckets(start: f64, factor: f64, length: u16) -> impl Iterato
         .take(length.into())
 }
 
+/// Exponential bucket distribution within a range
+/// /// Creates `length` buckets, where the lowest bucket is `min` and the highest bucket is `max`.
+/// /// The final +Inf bucket is not counted and not included in the returned iterator.
+/// /// The function panics if `length` is 0 or negative, or if `min` is 0 or negative.
+fn exponential_buckets_range(min: f64, max: f64, length: u16) -> impl Iterator<Item = f64> {
+    if length < 1 {
+        panic!("ExponentialBucketsRange length needs a positive length");
+    }
+    if min <= 0.0 {
+        panic!("ExponentialBucketsRange min needs to be greater than 0");
+    }
```

Comment:
We don't panic anywhere else in the crate. Reason being, that given that instrumentation is auxiliary only, folks likely don't want a panic. Would an empty Iterator work too?

Reply:
Sure! I have removed panics, but instead of an empty iterator I have added defaults with a warning. 
Empty iterator might have have side-effects in histogram functionality. 
Wdyt?